floris/floris_model.py

```diff
@@ -256,30 +263,31 @@ def _reinitialize(
         flow_field_dict = floris_dict["flow_field"]
         farm_dict = floris_dict["farm"]
 
-        # Make the given changes
-
-        # First check if wind data is not None,
-        # if not, get wind speeds, wind direction and
-        # turbulence intensity using the unpack_for_reinitialize
-        # method
-        if wind_data is not None:
-            if (
-                (wind_directions is not None)
-                or (wind_speeds is not None)
-                or (turbulence_intensities is not None)
-                or (heterogenous_inflow_config is not None)
-            ):
+        #
+        if (
+            (wind_directions is not None)
+            or (wind_speeds is not None)
+            or (turbulence_intensities is not None)
+            or (heterogenous_inflow_config is not None)
+        ):
+            if wind_data is not None:
                 raise ValueError(
                     "If wind_data is passed to reinitialize, then do not pass wind_directions, "
                     "wind_speeds, turbulence_intensities or "
                     "heterogenous_inflow_config as this is redundant"
                 )
-            (
-                wind_directions,
-                wind_speeds,
-                turbulence_intensities,
-                heterogenous_inflow_config,
-            ) = wind_data.unpack_for_reinitialize()
+            elif self.wind_data is not None:
```

Comment:
I think self.wind_data should be self._wind_data throughout floris_model

Reply:
I'm actually not sure about this---might need @rafmudaf 's input. My logic is that anywhere that the `wind_data` is modified (i.e. when it is set), it will set directly onto `self._wind_data`, but anywhere it is just called or used then it can use the getter version `self.wind_data`. But perhaps `self.wind_data` should be _only_ for external access, and all internal access (like the case you've pointed out here) should be `self._wind_data`.

Reply:
Confirmed with @rafmudaf that the current implementation is appropriate (no strong opinion either way, but this way if people read through the code and want to replicate parts, they'll see that they should be accessing the getter version